mne/io/eeglab/eeglab.py

```diff
@@ -654,8 +1000,22 @@ def read_events_eeglab(eeg, event_id=None, event_id_func='strip_to_integer',
 
     if isinstance(eeg, string_types):
         from scipy import io
-        eeg = io.loadmat(eeg, struct_as_record=False, squeeze_me=True,
-                         uint16_codec=uint16_codec)['EEG']
+
+        try:
+            fhandle = open(eeg, 'rb')
+            eeg = io.loadmat(fhandle, struct_as_record=False,
+                             squeeze_me=True,
+                             uint16_codec=uint16_codec)['EEG']
+            fhandle.close()
+        except NotImplementedError:
+            # Try to read new style Matlab file (Version 7.3+)
+            fhandle.close()
+            import h5py  # Added to read newer Matlab files (7.3 and later)
+            logger.info("Attempting to read Matlab style hdf file")
+            f = h5py.File(eeg)
+            eeg_dict = hdf_2_dict(f, f['EEG'], parent=None)
+            f.close()
+            eeg = _bunchify(eeg_dict)
```

Comment:
This looks redundant with code that is above. Can you refactor into a private function that is used both places?